File: update_videos.py
import re
import os

html_file = r'e:\01_emaus\02. Site\code.html'
videos_dir = r'e:\01_emaus\02. Site\EMAUS _ MIDIA\VIDEOS'

# get all video files
videos = []
for f in os.listdir(videos_dir):
    if f.lower().endswith(('.mov', '.mp4')):
        videos.append(f)

# natural sort
import re
def natural_keys(text):
    return [int(c) if c.isdigit() else c.lower() for c in re.split(r'(\d+)', text)]

videos.sort(key=natural_keys)

carousel_html = ""
for i, video in enumerate(videos, 1):
    src = f"EMAUS%20_%20MIDIA/VIDEOS/{video.replace(' ', '%20')}"
    # type attribute
    v_type = "video/mp4" if video.lower().endswith('.mp4') else "video/quicktime"
    
    slide = f"""            <!-- Video Slide {i} -->
            <div class="swiper-slide aspect-[9/16] md:aspect-[4/5] bg-stone-950 flex items-center justify-center overflow-hidden rounded-custom relative group">
              <video class="w-full h-full object-cover object-[center_top]" controls preload="none">
                <source src="{src}" type="{v_type}">
                Seu navegador não suporta o formato de vídeo.
              </video>
            </div>
"""
    carousel_html += slide

# Read HTML
with open(html_file, 'r', encoding='utf-8') as f:
    content = f.read()

# Replace between <div class="swiper-wrapper"> for videos-swiper
# Find <div class="swiper videos-swiper...
pattern = r'(<div class="swiper videos-swiper[^>]*>[\s\n]*<div class="swiper-wrapper">)[\s\S]*?(</div>[\s\n]*<!-- Add Pagination -->)'
replacement = r'\1\n' + carousel_html + r'          \2'
new_content = re.sub(pattern, replacement, content)

with open(html_file, 'w', encoding='utf-8') as f:
    f.write(new_content)

print(f"Updated {len(videos)} videos in HTML.")

File: update_videos.js
const fs = require('fs');

const htmlFile = 'e:\\01_emaus\\02. Site\\code.html';
const videosDir = 'e:\\01_emaus\\02. Site\\EMAUS _ MIDIA\\VIDEOS';

const files = fs.readdirSync(videosDir);
const videos = files.filter(f => f.toLowerCase().endsWith('.mov') || f.toLowerCase().endsWith('.mp4'));

// Natural sort
videos.sort((a, b) => {
  return a.localeCompare(b, undefined, { numeric: true, sensitivity: 'base' });
});

let carouselHtml = '';
videos.forEach((video, index) => {
  const i = index + 1;
  const src = `EMAUS%20_%20MIDIA/VIDEOS/${encodeURIComponent(video)}#t=0.001`;
  const ext = video.toLowerCase().split('.').pop();

  carouselHtml += `            <!-- Video Slide ${i} -->
            <div class="swiper-slide aspect-video bg-white rounded-custom p-2 sm:p-4 flex items-center justify-center shadow-sm border border-brand-beige">
              <video class="w-full h-full object-contain rounded bg-stone-950 swiper-no-swiping" controls preload="metadata" playsinline>
                <source src="${src}">
                Seu navegador não suporta o formato de vídeo.
              </video>
            </div>
`;
});

let content = fs.readFileSync(htmlFile, 'utf8');

const regex = /(<div class="swiper videos-swiper[^>]*>[\s\n]*<div class="swiper-wrapper">)[\s\S]*?(<\/div>[\s\n]*<!-- Add Pagination -->)/;
const newContent = content.replace(regex, `$1\n${carouselHtml}          $2`);

fs.writeFileSync(htmlFile, newContent, 'utf8');
console.log(`Updated ${videos.length} videos in HTML with node.`);
